components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java

```diff
@@ -391,52 +391,63 @@ public void updateTenant(TenantInfoBean tenantInfoBean) throws Exception {
+            TenantMgtUtil.addClaimsToUserStoreManager(tenant);
+
+            // filling the email value
+            if (tenantInfoBean.getEmail() != null && !tenantInfoBean.getEmail().equals("")) {
+                // validate the email
+                try {
+                    CommonUtil.validateEmail(tenantInfoBean.getEmail());
+                } catch (Exception e) {
+                    String msg = "Invalid email is provided.";
+                    log.error(msg, e);
+                    throw new Exception(msg, e);
+                }
+                tenant.setEmail(tenantInfoBean.getEmail());
+            }
 
-        boolean updatePassword = false;
-        if (tenantInfoBean.getAdminPassword() != null
-            && !tenantInfoBean.getAdminPassword().equals("")) {
-            updatePassword = true;
-        }
-        if (!userStoreManager.isReadOnly() && updatePassword) {
-            // now we will update the tenant admin with the admin given
-            // password.
+            UserRealm userRealm = configSystemRegistry.getUserRealm();
             try {
-                userStoreManager.updateCredentialByAdmin(tenantInfoBean.getAdmin(),
-                                                         tenantInfoBean.getAdminPassword());
+                userStoreManager = userRealm.getUserStoreManager();
             } catch (UserStoreException e) {
-                String msg = "Error in changing the tenant admin password, tenant domain: " +
-                             tenantInfoBean.getTenantDomain() + ". " + e.getMessage() + " for: " +
-                             tenantInfoBean.getAdmin();
+                String msg = "Error in getting the user store manager for tenant, tenant domain: " +
+                        tenantDomain + ".";
                 log.error(msg, e);
                 throw new Exception(msg, e);
             }
-        } else {
-            //Password should be empty since no password update done
-            tenantInfoBean.setAdminPassword("");
+
+            boolean updatePassword = false;
+            if (tenantInfoBean.getAdminPassword() != null
+                    && !tenantInfoBean.getAdminPassword().equals("")) {
+                updatePassword = true;
+            }
+            if (!userStoreManager.isReadOnly() && updatePassword) {
+                // now we will update the tenant admin with the admin given
+                // password.
+                try {
+                    userStoreManager.updateCredentialByAdmin(tenantInfoBean.getAdmin(),
+                            tenantInfoBean.getAdminPassword());
+                } catch (UserStoreException e) {
+                    String msg = "Error in changing the tenant admin password, tenant domain: " +
+                            tenantInfoBean.getTenantDomain() + ". " + e.getMessage() + " for: " +
+                            tenantInfoBean.getAdmin();
+                    log.error(msg, e);
+                    throw new Exception(msg, e);
```

Comment:
This code segment was already there. Due to formatting it is shown as a new addition.